tests/integration/api_v3/http/blocks.assets.test.js

```diff
@@ -0,0 +1,321 @@
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import moment from 'moment';
+
+const config = require('../../../config');
+const { api } = require('../../../helpers/api');
+
+const baseUrl = config.SERVICE_ENDPOINT;
+const baseUrlV3 = `${baseUrl}/api/v3`;
+const endpoint = `${baseUrlV3}/blocks/assets`;
+
+const {
+	goodRequestSchema,
+	badRequestSchema,
+	notFoundSchema,
+	wrongInputParamSchema,
+	metaSchema,
+} = require('../../../schemas/httpGenerics.schema');
+
+const {
+	blockAssetSchema,
+} = require('../../../schemas/api_v3/block.schema');
+
+describe('Blocks Assets API', () => {
+	let refBlockAsset;
+	beforeAll(async () => {
+		[refBlockAsset] = (await api.get(`${endpoint}?limit=1&offset=5`)).data;
+	});
+
+	describe('GET /blocks/assets', () => {
+		it('returns list of blocks assets', async () => {
+			const response = await api.get(`${endpoint}`);
+			expect(response).toMap(goodRequestSchema);
+			expect(response.data).toBeInstanceOf(Array);
+			expect(response.data.length).toBeGreaterThanOrEqual(1);
+			expect(response.data.length).toBeLessThanOrEqual(10);
+			response.data.forEach((blockAsset, i) => {
+				expect(blockAsset).toMap(blockAssetSchema);
+				if (i < response.data.length - 1) {
+					expect(blockAsset.block.height).toBe(response.data[i + 1].block.height + 1);
+				}
+			});
+			expect(response.meta).toMap(metaSchema);
+		});
+
+		it('returns list of blocks assets when call with limit 10', async () => {
+			const response = await api.get(`${endpoint}?limit=10`);
+			expect(response).toMap(goodRequestSchema);
+			expect(response.data).toBeInstanceOf(Array);
+			expect(response.data.length).toBeGreaterThanOrEqual(1);
+			expect(response.data.length).toBeLessThanOrEqual(10);
+			response.data.forEach((blockAsset, i) => {
+				expect(blockAsset).toMap(blockAssetSchema);
+				if (i < response.data.length - 1) {
+					expect(blockAsset.block.height).toBe(response.data[i + 1].block.height + 1);
+				}
+			});
+			expect(response.meta).toMap(metaSchema);
+		});
+
+		it('returns block assets by moduleID -> ok', async () => {
+			const response = await api.get(`${endpoint}?moduleID=${refBlockAsset.moduleID}`);
+			expect(response).toMap(goodRequestSchema);
+			response.data.forEach((blockAsset, i) => {
+				expect(blockAsset).toMap(blockAssetSchema);
+				if (i < response.data.length - 1) {
+					expect(blockAsset.block.height).toBe(response.data[i + 1].block.height + 1);
+				}
+			});
+			expect(response.meta).toMap(metaSchema);
+		});
+
+		it('returns block assets by blockID -> ok', async () => {
+			const response = await api.get(`${endpoint}?blockID=${refBlockAsset.block.id}`);
+			expect(response).toMap(goodRequestSchema);
+			response.data.forEach((blockAsset, i) => {
+				expect(blockAsset).toMap(blockAssetSchema);
+				if (i < response.data.length - 1) {
+					expect(blockAsset.block.height).toBe(response.data[i + 1].block.height + 1);
+				}
+			});
```

Comment:
This check can be removed. And, please enforce that the request parameters are respected in the response.
```suggestion
			response.data.forEach((blockAsset) => {
				expect(blockAsset).toMap(blockAssetSchema);
				expect(blockAsset.block.id).toBe(refBlockAsset.block.id);
			});
```